mirrord/layer/src/connection.rs

```diff
@@ -52,13 +56,8 @@ pub(crate) async fn connect(
             .connect(&progress)
             .await
             .unwrap_or_else(|err| handle_error(err, config))
-    } else if config.operator.enabled && let Some((operator_api, operator_ref)) =
-        OperatorApiDiscover::discover_operator(config, &progress).await
-    {
-        operator_api
-            .create_connection(operator_ref)
-            .await
-            .unwrap_or_else(|err| handle_error(err, config))
+    } else if config.operator && let Ok((operator_api, operator_ref)) = OperatorApi::discover(&config, &progress).await {
+        operator_api.create_connection(operator_ref).await.unwrap_or_else(|err| handle_operator_error(err))
```

Comment:
Why doesn't Discover already create the connection? Why does it return `operator_api` and `operator_ref`, only for `operator_ref` to be sent right back to `operator_api`?